sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/WindowTest.java

```diff
@@ -446,6 +447,55 @@ public void testDisplayData() {
   }
 
   @Test
+  public void testAssignDisplayData() {
+    FixedWindows windowFn = FixedWindows.of(Duration.standardHours(5));
+    AfterWatermark.FromEndOfWindow triggerBuilder = AfterWatermark.pastEndOfWindow();
+    Duration allowedLateness = Duration.standardMinutes(10);
+    Window.ClosingBehavior closingBehavior = ClosingBehavior.FIRE_ALWAYS;
+    OutputTimeFn<BoundedWindow> outputTimeFn = OutputTimeFns.outputAtLatestInputTimestamp();
+
+    WindowingStrategy<?, ?> updated =
+        WindowingStrategy.globalDefault()
+            .withWindowFn(windowFn)
+            .withTrigger(triggerBuilder)
+            .withMode(AccumulationMode.ACCUMULATING_FIRED_PANES)
+            .withAllowedLateness(allowedLateness)
+            .withClosingBehavior(closingBehavior)
+            .withOutputTimeFn(outputTimeFn);
+
+    DisplayData displayData = DisplayData.from(new Window.Assign<>(updated));
+
+    assertThat(displayData, hasDisplayItem("windowFn", windowFn.getClass()));
+    assertThat(displayData, includesDisplayDataFor("windowFn", windowFn));
+
+    assertThat(displayData, hasDisplayItem("trigger", triggerBuilder.toString()));
+    assertThat(displayData,
+        hasDisplayItem("accumulationMode", AccumulationMode.ACCUMULATING_FIRED_PANES.toString()));
+    assertThat(displayData,
+        hasDisplayItem("allowedLateness", allowedLateness));
+    assertThat(displayData, hasDisplayItem("closingBehavior", closingBehavior.toString()));
+    assertThat(displayData, hasDisplayItem("outputTimeFn"));
+  }
+
+  @Test
```

Comment:
Can you add a [DisplayDataEvalulator](https://github.com/apache/beam/blob/2eeeaa257c9336b2c48c3850c8508686a75be7b3/sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/display/DisplayDataEvaluator.java)-based test, similar to [AvroIOTest#testPrimitiveReadDisplayData()](https://github.com/apache/beam/blob/2eeeaa257c9336b2c48c3850c8508686a75be7b3/sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroIOTest.java#L415)? DisplayDataEvalulator tests that user-facing SDK transforms populate expected display data somewhere in its enclosed leaf transforms. This is valuable to ensure that we don't lose display data if a transform is refactored to be implemented by new leaf transforms. 

The tests are run across runners to ensure that any transform replacements by the runner also include expected display data (this may go away with new graph surgery work)